Yafc/Workspace/ProductionTable/ProductionLinkSummaryScreen.cs

```diff
@@ -51,67 +51,70 @@ private void BuildScrollArea(ImGui gui) {
+            if ((row.FindLink(link.goods, out var otherLink) && otherLink != link)) {
+                if (!relationLinks.ContainsKey(otherLink)) {
+                    relationLinks.Add(otherLink, []);
                 }
-                else {
-                    table = unlinked;
-                }
-
-                table.Add((row, localFlow));
+                table = relationLinks[otherLink];
             }
-            var color = 1;
-            gui.spacing = 0.75f;
-            if (childLinks.Values.Any(e => e.Any())) {
-                gui.BuildText("Child links: ", Font.productionTableHeader);
-                foreach (var relTable in childLinks.Values) {
-                    BuildFlow(gui, relTable, relTable.Sum(e => Math.Abs(e.flow)), false, color++);
-                }
+            else {
+                table = unlinked;
             }
-            if (parentLinks.Values.Any(e => e.Any())) {
-                gui.BuildText("Parent links: ", Font.productionTableHeader);
-                foreach (var relTable in parentLinks.Values) {
-                    BuildFlow(gui, relTable, relTable.Sum(e => Math.Abs(e.flow)), false, color++);
-                }
+
+            table.Add((row, localFlow));
+        }
+        var color = 1;
+        gui.spacing = 0.75f;
+        if (childLinks.Values.Any(e => e.Any())) {
+            gui.BuildText("Child links: ", Font.productionTableHeader);
+            foreach (var relTable in childLinks.Values) {
+                BuildFlow(gui, relTable, relTable.Sum(e => Math.Abs(e.flow)), false, color++);
             }
-            if (otherLinks.Values.Any(e => e.Any())) {
-                gui.BuildText("Unrelated links: ", Font.productionTableHeader);
-                foreach (var relTable in otherLinks.Values) {
-                    BuildFlow(gui, relTable, relTable.Sum(e => Math.Abs(e.flow)), false, color++);
-                }
+        }
+        if (parentLinks.Values.Any(e => e.Any())) {
+            gui.BuildText("Parent links: ", Font.productionTableHeader);
+            foreach (var relTable in parentLinks.Values) {
+                BuildFlow(gui, relTable, relTable.Sum(e => Math.Abs(e.flow)), false, color++);
             }
-            if (unlinked.Any()) {
-                gui.BuildText("Unlinked: ", Font.subheader);
-                BuildFlow(gui, unlinked, 0, false);
+        }
+        if (otherLinks.Values.Any(e => e.Any())) {
+            gui.BuildText("Unrelated links: ", Font.productionTableHeader);
+            foreach (var relTable in otherLinks.Values) {
+                BuildFlow(gui, relTable, relTable.Sum(e => Math.Abs(e.flow)), false, color++);
+            }
+        }
+        if (unlinked.Any()) {
+            gui.BuildText("Unlinked: ", Font.subheader);
+            BuildFlow(gui, unlinked, 0, false);
+        }
+    }
+
+    public static ProductionTable? FindProductionTable(ModelObject link, out List<ModelObject> parents) {
```

Comment:
The possibly-null checker is still unhappy (on line 58), because this is returning `ProductionTable?`, not `ProductionTable`.

```suggestion
    public static ProductionTable FindProductionTable(ModelObject link, out List<ModelObject> parents) {
```

Reply:
Yeah drat, i removed the casting but forgot i  need to keep the nullcheck. Added it back.